### PR TITLE
spacy 3.8.2

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Destination channel:** main

### Links

- [PKG-5891](https://anaconda.atlassian.net/browse/PKG-5891) 
- [Upstream repository](https://github.com/explosion/spaCy/tree/release-v3.8.2)
- Requirements:
  - https://github.com/explosion/spaCy/blob/release-v3.8.2/pyproject.toml
  - https://github.com/explosion/spaCy/blob/release-v3.8.2/setup.cfg 
  - https://github.com/explosion/spaCy/blob/release-v3.8.2/setup.py

### Explanation of changes:

- Skip python <3.9 because of numpy >=2
- Do not use `{{ pin_compatible('numpy') }}` in `run` because it will pick up an incompatible numpy 1.2x
- Update dependencies and pinnings

### Notes:

- It requires updating:
  - https://github.com/AnacondaRecipes/cython-blis-feedstock/pull/6
  - https://github.com/AnacondaRecipes/thinc-feedstock/pull/17
  - https://github.com/AnacondaRecipes/cython-feedstock/pull/22

[PKG-5891]: https://anaconda.atlassian.net/browse/PKG-5891?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
An error on `osx-64` and `linux-64`:
```
  spacy/matcher/levenshtein.c:2248:3: error: no member named 'use_tracing' in 'struct _PyCFrame'
    __Pyx_TraceCall("make_levenshtein_compare", __pyx_f[0], 31, 0, __PYX_ERR(0, 31, __pyx_L1_error));
```

**Comment by skupr-anaconda** [comment]
> An error on `osx-64` and `linux-64`:
> 
> ```
>   spacy/matcher/levenshtein.c:2248:3: error: no member named 'use_tracing' in 'struct _PyCFrame'
>     __Pyx_TraceCall("make_levenshtein_compare", __pyx_f[0], 31, 0, __PYX_ERR(0, 31, __pyx_L1_error));
> ```

It looks like `cython 0.29.37` can fix the issue, see https://github.com/explosion/spaCy/issues/13293#issuecomment-2373067037. We have `cython 0.29.36` on the `main` channel.

**Comment by skupr-anaconda** [comment]
Yup, it fails with **cython 3** in `host`:
```
File "/croot/spacy_1728486691614/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.12/site-packages/Cython/Build/Dependencies.py", line 1321, in cythonize_one

      raise CompileError(None, pyx_file)
  Cython.Compiler.Errors.CompileError: spacy/ml/parser_model.pyx
```

**Comment by skupr-anaconda** [comment]
> > An error on `osx-64` and `linux-64`:
> > ```
> >   spacy/matcher/levenshtein.c:2248:3: error: no member named 'use_tracing' in 'struct _PyCFrame'
> >     __Pyx_TraceCall("make_levenshtein_compare", __pyx_f[0], 31, 0, __PYX_ERR(0, 31, __pyx_L1_error));
> > ```
> 
> It looks like `cython 0.29.37` can fix the issue, see [explosion/spaCy#13293 (comment)](https://github.com/explosion/spaCy/issues/13293#issuecomment-2373067037). We have `cython 0.29.36` on the `main` channel.

Yes, using `cython 0.29.37` fixes the issue. See the cython Changelog https://cython.readthedocs.io/en/0.29.x/src/changes.html#id1

**Review by SC426 — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -27,31 +28,30 @@ requirements:
   host:
     - pip
     - python
-    - numpy {{ numpy }}
```
Comment (psteyer) [inline recipe/meta.yaml:30]: Out of curiosity, why did we need to change this from the `cbc.yaml` `numpy` version to a pinned version?

Comment (skupr-anaconda) [reply recipe/meta.yaml:30]: spacy and a few of its dependencies require numpy >=2,<3 to build the package. Our base cbc.yaml pins numpy to 1.2x versions but they are incompatible

**Review by ELundby45 — APPROVED** [review]

**Pull request merged**